src/components/ebay-phone-input/component.ts

```diff
@@ -0,0 +1,132 @@
+import type { WithNormalizedProps } from "../../global";
+import type { Input as MenuButtonInput } from "../ebay-menu-button/component";
+import type { default as Textbox } from "../ebay-textbox/component-browser";
+import mask, { stripNonDigits } from "../../common/mask";
+
+import countries, {
+    type CountriesLanguageInterface,
+    type CountryInterface,
+    countriesEnglish,
+} from "../../common/countries";
+import { AttrString } from "marko/tags-html";
+
+export interface CountryNames {
+    [index: string]: string;
+}
+
+export interface PhoneInputEvent {
+    originalEvent?: Event;
+    value?: string;
+    rawValue?: string;
+    callingCode?: string;
+    countryCode?: string;
+}
+
+interface PhoneInputInput extends Omit<MenuButtonInput, `on${string}`> {
+    disabled?: boolean;
+    "country-names"?: CountriesLanguageInterface;
+    "a11y-button-text"?: AttrString;
+    "a11y-menu-text"?: MenuButtonInput["a11yText"];
+    "floating-label"?: string;
+    readonly?: boolean;
+    invalid?: boolean;
+    "country-code"?: string;
+    value?: string;
+    "on-keyup"?: (event: PhoneInputEvent) => void;
+    "on-keypress"?: (event: PhoneInputEvent) => void;
+    "on-keydown"?: (event: PhoneInputEvent) => void;
+    "on-focus"?: (event: PhoneInputEvent) => void;
+    "on-blur"?: (event: PhoneInputEvent) => void;
+    "on-collapse"?: (event: PhoneInputEvent) => void;
+    "on-expand"?: (event: PhoneInputEvent) => void;
+    "on-change"?: (event: PhoneInputEvent) => void;
+}
+
+export interface Input extends WithNormalizedProps<PhoneInputInput> {}
+
+export interface State {
+    index: number;
+}
+
+class PhoneInput extends Marko.Component<Input, State> {
+    mask: any;
+
+    onCreate() {
+        this.state = {
+            index: 0,
+        };
+    }
+
+    get selectedCountry(): CountryInterface {
```

Comment:
My opinion so it doesn't have to change, but I don't love getters like this because they hide the fact that you are doing work wherever they are called.